### PR TITLE
VIH-9382 Changes for filtering by CSO

### PR DESCRIPTION
**Pull request by oliver-scott**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-9832
https://tools.hmcts.net/jira/browse/VIH-9834

### Change description ###
Applies changes to the venue + cso lists on the hearing selection screen for VHOs.
- Currently, filtering hearings by cso returns the hearings that are in the venues to which the selected csos have been allocated to. Consequently this can show hearings assigned to other csos in addition to the selected one. To remedy this, filter the conferences specifically allocated to the csos, instead of fetching the venues.
- When returning to the hearing selection screen, pre-populate the csos that had been previously filtered on, rather than the venues associated with those csos.
- Some changes have made to the mappers (ConferenceForAdminResponse -> ConferenceForVhOfficerResponse) to make the allocated cso properties available to filter on.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/src/app/shared/venue-list/vh-officer-venue-list/vh-officer-venue-list.component.ts [hunk 1]
```diff
@@ -68,4 +94,22 @@ export class VhOfficerVenueListComponent extends VenueListComponentDirective imp
     get showVhoSpecificContent(): boolean {
         return true;
     }
+
+    async updateCsoFilterSelection(filter: CsoFilter) {
+        const selectCso = (csoId: string) => {
+            this.selectedCsos = [...this.selectedCsos, csoId];
+        };
```
Comment (will-craig) [inline VideoWeb/VideoWeb/ClientApp/src/app/shared/venue-list/vh-officer-venue-list/vh-officer-venue-list.component.ts:101]: is this function selectCso neccessary? Would doing `this.selectedCsos.push(VhOfficerVenueListComponent.ALLOCATED_TO_ME)` 
`this.selectedCsos.push(id)` 
`this.selectedCsos.push(VhOfficerVenueListComponent.UNALLOCATED)`

on lines 106 and 109 and 112 not achieve the same thing?

**Review by will-craig — COMMENTED** [review]
Looks great, good job. One question below

File: VideoWeb/VideoWeb/ClientApp/src/app/shared/venue-list/vh-officer-venue-list/vh-officer-venue-list.component.ts (hunk 1 above)
Comment (oliver-scott) [reply VideoWeb/VideoWeb/ClientApp/src/app/shared/venue-list/vh-officer-venue-list/vh-officer-venue-list.component.ts:101]: thanks. Yes that's a good shout. I probably over-abstracted it with that function

**Review by will-craig — APPROVED** [review]

**Review by marcogagliardi — APPROVED** [review]
👌🏽

File: VideoWeb/VideoWeb/ClientApp/src/app/shared/venue-list/vh-officer-venue-list/vh-officer-venue-list.component.ts (hunk 1 above)
Comment (oliver-scott) [reply VideoWeb/VideoWeb/ClientApp/src/app/shared/venue-list/vh-officer-venue-list/vh-officer-venue-list.component.ts:101]: I've had to revert this change as it broke the pre-populating of the dropdown list when navigating back to the page.

I remember why I did it that way now - doing .push does not trigger change detection for the dropdown list - https://github.com/ng-select/ng-select#change-detection

**Review by marcogagliardi — APPROVED** [review]
👌🏽

**Review by komaldabhi1509 — APPROVED** [review]
Test Approved

**Review by shaed-parkar — APPROVED** [review]

**Pull request merged**